server/s3_common_utilities.cc

```diff
@@ -126,6 +126,52 @@ bool stoi(const std::string &str, int &value) {
   return isvalid;
 }
 
+void size_based_bucketing_of_objects(std::string &oid_str, size_t obj_size){
+  s3_log(S3_LOG_INFO, ""  , "Entering\n");
+  size_t object_size_for_bucketing = obj_size;
+
+  // object size range                           prefix
+
+  // size > 100GB                                 'D'
+  // 1GB < size <= 100GB                          'E'
+  // 50MB < size <= 1GB                           'F'
+  // 10KB < size <= 50MB                          'G'
+  // 1KB < size <= 10KB                           'H'
+  // size <= 1KB                                  'I'
+
+  // NOTE : chars A/B/C have been left out for objects bigger than 100GB
+  // Listing of keys in motr happens in lexicographic order, hence bigger object
+  // will get listed first.
+
+  switch(object_size_for_bucketing){
+       case 0 ... 1024:
+            oid_str = 'I' + oid_str;      // less than or equal to 1KB
```

Comment:
better to use string::apend() function instead of assignment operator, that would be more efficient.